iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -1195,6 +1195,27 @@ public boolean isBoundaryElement(Identifiable<?> identifiable) {
         return identifiable.getType() == IdentifiableType.DANGLING_LINE && !((DanglingLine) identifiable).isPaired();
     }
 
+    @Override
+    public void flatten() {
+        if (subnetworks.isEmpty()) {
+            // Nothing to do
+            return;
+        }
+        subnetworks.values().forEach(subnetwork -> {
+            // The subnetwork ref chain should point to the current network's subnetworkRef.
+            // This way, all its network elements (using this ref chain) will have a reference to the current network
```

Comment:
One thing I missed at first was that the reference is through a double RefChain. Is it worth mentioning it or would it make it more confusing?

Reply:
I added more details in the comment.